editor/src/clj/editor/gui.clj

```diff
@@ -1783,22 +1809,24 @@
 
   (display-order (into base-display-order [:manual-size :enabled :visible :text :line-break :font :material :color :alpha :inherit-alpha :text-leading :text-tracking :outline :outline-alpha :shadow :shadow-alpha :layer]))
 
-  (output font-data font/FontData (g/fnk [font-datas font]
-                                    (or (font-datas font)
-                                        (font-datas ""))))
+  (output font-data font/FontData (g/fnk [costly-gui-scene-info font]
+                                    (let [font-datas (:font-datas costly-gui-scene-info)]
+                                      (or (get font-datas font)
+                                          (get font-datas "")))))
 
   ;; Overloaded outputs
   (output node-msg g/Any :cached produce-text-node-msg)
   (output gpu-texture TextureLifecycle (g/fnk [font-data] (:texture font-data)))
   (output scene-renderable-user-data g/Any :cached
-          (g/fnk [manual-size font font-shaders material material-shaders pivot text-data color+alpha]
+          (g/fnk [costly-gui-scene-info manual-size font material material-shader pivot text-data color+alpha]
                  (let [[w h] manual-size
                        offset (pivot-offset pivot manual-size)
                        lines (mapv conj (apply concat (take 4 (partition 2 1 (cycle (geom/transl offset [[0 0] [w 0] [w h] [0 h]]))))) (repeat 0))
                        font-map (get-in text-data [:font-data :font-map])
                        texture-recip-uniform (font/get-texture-recip-uniform font-map)
-                       material-shader (when (not (empty? material)) (material-shaders material))
-                       font-shader (or material-shader (font-shaders font) (font-shaders ""))
+                       material-shader (when (not (empty? material)) material-shader)
+                       font-shaders (:font-shaders costly-gui-scene-info)
+                       font-shader (or material-shader (get font-shaders font) (get font-shaders ""))
```

Comment:
The `material-shader` overrides the shader associated with the font if specified for this node.